api_app/analyzers_manager/file_analyzers/filescan.py

```diff
@@ -0,0 +1,110 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+import time
+
+import requests
+
+from api_app.analyzers_manager.classes import FileAnalyzer
+from api_app.exceptions import AnalyzerRunException
+from tests.mock_utils import MockResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class FileScan(FileAnalyzer):
+    def set_params(self, params):
+        self.upload_file = params.get("upload_file", True)
+        self.session = requests.Session()
+        self.request_url = "https://www.filescan.io/"
+        self.task_id = 0
+        self.poll_distance = 10
+        self.result = {}
+        self.max_post_tries = params.get("max_post_tries", 5)
+        self.max_get_tries = 30
+
+    def run(self):
+        binary = self.read_file_bytes()
+        if not binary:
+            raise AnalyzerRunException("File is empty")
+        self.__filescan_request_scan(binary)
+        self.__poll_status()
+        result = self.__fetch_and_build_result()
+        return result
+
+    def __filescan_request_scan(self, binary):
+        logger.info(f"Requesting Scan for: ({self.filename}), {self.md5}")
+
+        name_to_send = self.filename if self.upload_file else self.md5
+        files = {"file": (name_to_send, binary)}
+        post_sucess = False
+        for chance in range(self.max_post_tries):
+            logger.info(
+                f"#{chance} for file analysis  of ({self.filename}), {self.md5}"
+            )
+            response = self.session.post(
+                self.request_url + "api/scan/file", files=files
+            )
+            upl = response.status_code
+            logger.info(f"UPLOAD CODE: {upl}")
+            if response.status_code != 200:
+                logger.info(f"Error: {response.status_code}")
+                time.sleep(5)
+                continue
+            else:
+                post_sucess = True
+                break
+        if post_sucess:
+            json_response = response.json()
+            self.task_id = json_response["flow_id"]
+            # self.task_id = "619f8affb71ceed41530067b"
+            logger.info(f"TASK ID: {self.task_id}")
+        else:
+            raise AnalyzerRunException(
+                "failed max tries to post file to Filescan for analysis"
+            )
+
+    def __poll_status(self):
+        for chance in range(self.max_get_tries):
+            time.sleep(self.poll_distance)
+            url = self.request_url + "api/scan/" + str(self.task_id) + "/report"
+            logger.info(f"Polling #try{chance+1}")
+            response = self.session.get(url)
+            json_response = response.json()
+            status = json_response.get("allFinished")
+            logger.info(f"CURRENT STATUS: {status}")
+            if str(status) == "True":
```

Comment:
I assume that `allFinished` is a boolean already so type-casting to str is a bad idea.

Reply:
Nope, as `allFinished` is being retrieved from JSON Obj it's of type string
I tried running it without typecasting, it wasn't hitting if condition even though the condition is attained due to differences in their types

Reply:
![image](https://user-images.githubusercontent.com/16389167/144010569-1f65535a-d813-4ec5-90e3-4bb44300a257.png)
that's boolean, isn't it ?

Reply:
so `if json_response["allFinished"]:` should be enough.

Reply:
Yep This one works